src/test/unit/io/deserializer_test.cpp

```diff
@@ -0,0 +1,899 @@
+#include <stan/io/deserializer.hpp>
+// expect_near_rel comes from lib/stan_math
+#include <test/unit/math/expect_near_rel.hpp>
+#include <gtest/gtest.h>
+
+TEST(deserializer_scalar, read) {
```

Comment:
I think the only major tests missing are if we took this file and replaced all the `double` s with `stan::math::var`. That will be the main way this code is used so probably worth it.

Reply:
I'm gonna do this real quick actually.

Reply:
I added them in `deserializer_var_test.cpp`. Now that I copy-pasted them there I'm not sure how much value they add. Maybe just needed a couple `read` tests to make sure that things compiled rather than the full gamut of read_lb, read_ub, read_corr_matrix, etc.